datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -836,7 +836,10 @@ fn add_roundrobin_on_top(
     n_target: usize,
 ) -> Result<DistributionContext> {
     // Adding repartition is helpful:
```

Comment:
```suggestion
    // Always perform repartitioning on data sources, as they may be imbalanced or consist of too many small partitions.
```
A 'why' comment here can help understanding